Build issue with `universal2` on macos

**Describe the bug**
On macos, building a `universal2` build  (`-arch arm64 -arch x86_64` options) of `c-blosc2` fails when enabling `SHUFFLE_[SSE2/AVX2]_ENABLED` as those options are activated both for `x86_64` and `arm64`. I expect similar issue the other way around with `SHUFFLE_NEON_ENABLED`.

Related to similar issue on `c-blosc`: https://github.com/Blosc/c-blosc/issues/334

**To Reproduce**
On a macos ARM computer with a `universal2` build of Python, compile `blosc2` HDF5 filter as part of `hdf5plugin` (blosc2 PR: https://github.com/silx-kit/hdf5plugin/pull/201) with `python setup.py build`

**Expected behavior**
Even if `SHUFFLE_[SSE2/AVX2/NEON]_ENABLED` are set, if the target platform does not support those instructions they should not be considered.
I would expect the same behavior as proposed for blosc(1) in PR https://github.com/Blosc/c-blosc/pull/347.

**Logs**

<details><summary>Compilation logs...</summary>
```
gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch arm64 -arch x86_64 -g -DHAVE_LZ4=1 -DHAVE_ZLIB=1 -DHAVE_ZSTD=1 -DH5_USE_18_API -DSHUFFLE_SSE2_ENABLED=1 -Isrc/hdf5/include -Isrc/hdf5/include/darwin -Isrc/c-blosc2 -Isrc/c-blosc2/blosc -Isrc/c-blosc2/include -Isrc/c-blosc/internal-complibs/lz4-1.9.3 -Isrc/c-blosc/internal-complibs/zlib-1.2.11 -Isrc/c-blosc/internal-complibs/zstd-1.5.2 -Isrc/c-blosc/internal-complibs/zstd-1.5.2/common -Isrc/PyTables/hdf5-blosc2/src -Ivenv/py39/include -I/Library/Frameworks/Python.framework/Versions/3.9/include/python3.9 -c src/c-blosc2/blosc/bitshuffle-sse2.c -o build/temp.macosx-10.9-universal2-3.9/src/c-blosc2/blosc/bitshuffle-sse2.o -std=gnu99 -O3 -ffast-math -pthread -mcpu=native
src/c-blosc2/blosc/bitshuffle-sse2.c:29:4: error: SSE2 is not supported by the target architecture/platform and/or this compiler.
  #error SSE2 is not supported by the target architecture/platform and/or this compiler.
   ^
In file included from src/c-blosc2/blosc/bitshuffle-sse2.c:32:
In file included from /Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include/emmintrin.h:13:
In file included from /Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include/xmmintrin.h:13:
/Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include/mmintrin.h:50:12: error: invalid conversion between vector type '__m64' (vector of 1 'long long' value) and integer type 'int' of different size
    return (__m64)__builtin_ia32_vec_init_v2si(__i, 0);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include/mmintrin.h:129:12: error: invalid conversion between vector type '__m64' (vector of 1 'long long' value) and integer type 'int' of different size
    return (__m64)__builtin_ia32_packsswb((__v4hi)__m1, (__v4hi)__m2);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include/mmintrin.h:159:12: error: invalid conversion between vector type '__m64' (vector of 1 'long long' value) and integer type 'int' of different size
    return (__m64)__builtin_ia32_packssdw((__v2si)__m1, (__v2si)__m2);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include/mmintrin.h:189:12: error: invalid conversion between vector type '__m64' (vector of 1 'long long' value) and integer type 'int' of different size
    return (__m64)__builtin_ia32_packuswb((__v4hi)__m1, (__v4hi)__m2);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include/mmintrin.h:216:12: error: invalid conversion between vector type '__m64' (vector of 1 'long long' value) and integer type 'int' of different size
    return (__m64)__builtin_ia32_punpckhbw((__v8qi)__m1, (__v8qi)__m2);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include/mmintrin.h:239:12: error: invalid conversion between vector type '__m64' (vector of 1 'long long' value) and integer type 'int' of different size
    return (__m64)__builtin_ia32_punpckhwd((__v4hi)__m1, (__v4hi)__m2);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include/mmintrin.h:260:12: error: invalid conversion between vector type '__m64' (vector of 1 'long long' value) and integer type 'int' of different size
    return (__m64)__builtin_ia32_punpckhdq((__v2si)__m1, (__v2si)__m2);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include/mmintrin.h:287:12: error: invalid conversion between vector type '__m64' (vector of 1 'long long' value) and integer type 'int' of different size
    return (__m64)__builtin_ia32_punpcklbw((__v8qi)__m1, (__v8qi)__m2);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include/mmintrin.h:310:12: error: invalid conversion between vector type '__m64' (vector of 1 'long long' value) and integer type 'int' of different size
    return (__m64)__builtin_ia32_punpcklwd((__v4hi)__m1, (__v4hi)__m2);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include/mmintrin.h:331:12: error: invalid conversion between vector type '__m64' (vector of 1 'long long' value) and integer type 'int' of different size
    return (__m64)__builtin_ia32_punpckldq((__v2si)__m1, (__v2si)__m2);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include/mmintrin.h:352:12: error: invalid conversion between vector type '__m64' (vector of 1 'long long' value) and integer type 'int' of different size
    return (__m64)__builtin_ia32_paddb((__v8qi)__m1, (__v8qi)__m2);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include/mmintrin.h:373:12: error: invalid conversion between vector type '__m64' (vector of 1 'long long' value) and integer type 'int' of different size
    return (__m64)__builtin_ia32_paddw((__v4hi)__m1, (__v4hi)__m2);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include/mmintrin.h:394:12: error: invalid conversion between vector type '__m64' (vector of 1 'long long' value) and integer type 'int' of different size
    return (__m64)__builtin_ia32_paddd((__v2si)__m1, (__v2si)__m2);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include/mmintrin.h:416:12: error: invalid conversion between vector type '__m64' (vector of 1 'long long' value) and integer type 'int' of different size
    return (__m64)__builtin_ia32_paddsb((__v8qi)__m1, (__v8qi)__m2);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include/mmintrin.h:439:12: error: invalid conversion between vector type '__m64' (vector of 1 'long long' value) and integer type 'int' of different size
    return (__m64)__builtin_ia32_paddsw((__v4hi)__m1, (__v4hi)__m2);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include/mmintrin.h:461:12: error: invalid conversion between vector type '__m64' (vector of 1 'long long' value) and integer type 'int' of different size
    return (__m64)__builtin_ia32_paddusb((__v8qi)__m1, (__v8qi)__m2);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include/mmintrin.h:483:12: error: invalid conversion between vector type '__m64' (vector of 1 'long long' value) and integer type 'int' of different size
    return (__m64)__builtin_ia32_paddusw((__v4hi)__m1, (__v4hi)__m2);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/lib/clang/13.0.0/include/mmintrin.h:504:12: error: invalid conversion between vector type '__m64' (vector of 1 'long long' value) and integer type 'int' of different size
    return (__m64)__builtin_ia32_psubb((__v8qi)__m1, (__v8qi)__m2);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
```

</details>

**System information:**
 - OS: macos
 - Compiler clang
 - Version 2.6.0

--- a/blosc/bitshuffle-altivec.c
+++ b/blosc/bitshuffle-altivec.c
@@ -26,9 +26,8 @@
 #include "bitshuffle-altivec.h"
 
 /* Make sure ALTIVEC is available for the compilation target and compiler. */
-#if !defined(__ALTIVEC__)
-  #error ALTIVEC is not supported by the target architecture/platform and/or this compiler.
-#endif
+#if defined(__ALTIVEC__)
+
 #include <altivec.h>
 #include "transpose-altivec.h"
 
@@ -590,3 +589,5 @@ int64_t bshuf_untrans_bit_elem_altivec(void* in, void* out, const size_t size,
 
   return count;
 }
+
+#endif /* defined(__ALTIVEC__) */

--- a/blosc/bitshuffle-avx2.c
+++ b/blosc/bitshuffle-avx2.c
@@ -27,9 +27,7 @@
 
 
 /* Make sure AVX2 is available for the compilation target and compiler. */
-#if !defined(__AVX2__)
-  #error AVX2 is not supported by the target architecture/platform and/or this compiler.
-#endif
+#if defined(__AVX2__)
 
 #include <immintrin.h>
 
@@ -255,3 +253,5 @@ int64_t bshuf_untrans_bit_elem_avx2(void* in, void* out, const size_t size,
 
   return count;
 }
+
+#endif /* defined(__AVX2__) */

--- a/blosc/bitshuffle-neon.c
+++ b/blosc/bitshuffle-neon.c
@@ -12,9 +12,7 @@
 #include "bitshuffle-neon.h"
 
 /* Make sure NEON is available for the compilation target and compiler. */
-#if !defined(__ARM_NEON)
-  #error NEON is not supported by the target architecture/platform and/or this compiler.
-#endif
+#if defined(__ARM_NEON)
 
 #include <arm_neon.h>
 
@@ -1000,6 +998,8 @@ bitunshuffle_neon(void* _src, void* _dest, const size_t size,
          so we're done processing here. */
       return count;
   }
-  
+
   return (int64_t)size * (int64_t)elem_size;
 }
+
+#endif /* defined(__ARM_NEON) */

--- a/blosc/bitshuffle-sse2.c
+++ b/blosc/bitshuffle-sse2.c
@@ -25,9 +25,7 @@
 #include "bitshuffle-sse2.h"
 
 /* Make sure SSE2 is available for the compilation target and compiler. */
-#if !defined(__SSE2__)
-  #error SSE2 is not supported by the target architecture/platform and/or this compiler.
-#endif
+#if defined(__SSE2__)
 
 #include <emmintrin.h>
 
@@ -474,3 +472,5 @@ int64_t bshuf_untrans_bit_elem_sse2(void* in, void* out, const size_t size,
 
   return count;
 }
+
+#endif /* defined(__SSE2__) */

--- a/blosc/shuffle-altivec.c
+++ b/blosc/shuffle-altivec.c
@@ -12,9 +12,7 @@
 #include "shuffle-altivec.h"
 
 /* Make sure ALTIVEC is available for the compilation target and compiler. */
-#if !defined(__ALTIVEC__)
-  #error ALTIVEC is not supported by the target architecture/platform and/or this compiler.
-#endif
+#if defined(__ALTIVEC__)
 
 #include <altivec.h>
 #include "transpose-altivec.h"
@@ -421,3 +419,5 @@ unshuffle_altivec(const int32_t bytesoftype, const int32_t blocksize,
     unshuffle_generic_inline(bytesoftype, vectorizable_bytes, blocksize, _src, _dest);
   }
 }
+
+#endif /* defined(__ALTIVEC__) */

--- a/blosc/shuffle-avx2.c
+++ b/blosc/shuffle-avx2.c
@@ -12,9 +12,7 @@
 #include "shuffle-avx2.h"
 
 /* Make sure AVX2 is available for the compilation target and compiler. */
-#if !defined(__AVX2__)
-  #error AVX2 is not supported by the target architecture/platform and/or this compiler.
-#endif
+#if defined(__AVX2__)
 
 #include <immintrin.h>
 
@@ -745,3 +743,5 @@ unshuffle_avx2(const int32_t bytesoftype, const int32_t blocksize,
     unshuffle_generic_inline(bytesoftype, vectorizable_bytes, blocksize, _src, _dest);
   }
 }
+
+#endif /* defined(__AVX2__) */

--- a/blosc/shuffle-neon.c
+++ b/blosc/shuffle-neon.c
@@ -13,9 +13,7 @@
 #include "shuffle-neon.h"
 
 /* Make sure NEON is available for the compilation target and compiler. */
-#if !defined(__ARM_NEON)
-#error NEON is not supported by the target architecture/platform and/or this compiler.
-#endif
+#if defined(__ARM_NEON)
 
 #include <arm_neon.h>
 
@@ -414,3 +412,5 @@ unshuffle_neon(const int32_t bytesoftype, const int32_t blocksize,
         unshuffle_generic_inline(bytesoftype, vectorizable_bytes, blocksize, _src, _dest);
     }
 }
+
+#endif /* defined(__ARM_NEON) */

--- a/blosc/shuffle-sse2.c
+++ b/blosc/shuffle-sse2.c
@@ -12,9 +12,7 @@
 #include "shuffle-sse2.h"
 
 /* Make sure SSE2 is available for the compilation target and compiler. */
-#if !defined(__SSE2__)
-  #error SSE2 is not supported by the target architecture/platform and/or this compiler.
-#endif
+#if defined(__SSE2__)
 
 #include <emmintrin.h>
 
@@ -615,3 +613,5 @@ unshuffle_sse2(const int32_t bytesoftype, const int32_t blocksize,
     unshuffle_generic_inline(bytesoftype, vectorizable_bytes, blocksize, _src, _dest);
   }
 }
+
+#endif /* defined(__SSE2__) */

--- a/blosc/shuffle.c
+++ b/blosc/shuffle.c
@@ -141,7 +141,7 @@ __cpuidex(int32_t cpuInfo[4], int32_t function_id, int32_t subfunction_id) {
 
 // GCC folks added _xgetbv in immintrin.h starting in GCC 9
 // See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=71659
-#if !(defined(_IMMINTRIN_H_INCLUDED) && (BLOSC_GCC_VERSION >= 900))
+#if defined(__i386__) && !(defined(_IMMINTRIN_H_INCLUDED) && (BLOSC_GCC_VERSION >= 900))
 /* Reads the content of an extended control register.
    https://software.intel.com/en-us/articles/how-to-detect-new-instruction-support-in-the-4th-generation-intel-core-processor-family
 */
@@ -160,7 +160,7 @@ _xgetbv(uint32_t xcr) {
     );
   return ((uint64_t)edx << 32) | eax;
 }
-#endif  // !(defined(_IMMINTRIN_H_INCLUDED) && (BLOSC_GCC_VERSION >= 900))
+#endif  // defined(__i386__) && !(defined(_IMMINTRIN_H_INCLUDED) && (BLOSC_GCC_VERSION >= 900))
 #endif /* defined(_MSC_VER) */
 
 #ifndef _XCR_XFEATURE_ENABLED_MASK
@@ -204,7 +204,7 @@ static blosc_cpu_features blosc_get_cpu_features(void) {
   bool ymm_state_enabled = false;
   //bool zmm_state_enabled = false;  // commented this out for avoiding an 'unused variable' warning
 
-#if defined(_XCR_XFEATURE_ENABLED_MASK)
+#if defined(__i386__) && defined(_XCR_XFEATURE_ENABLED_MASK)
   if (xsave_available && xsave_enabled_by_os && (
       sse2_available || sse3_available || ssse3_available
       || sse41_available || sse42_available
@@ -219,7 +219,7 @@ static blosc_cpu_features blosc_get_cpu_features(void) {
         restored as well as all of zmm16-zmm31 and the opmask registers. */
     //zmm_state_enabled = (xcr0_contents & 0x70) == 0x70;
   }
-#endif /* defined(_XCR_XFEATURE_ENABLED_MASK) */
+#endif /* defined(__i386__) && defined(_XCR_XFEATURE_ENABLED_MASK) */
 
 #if defined(BLOSC_DUMP_CPU_INFO)
   printf("Shuffle CPU Information:\n");

--- a/blosc/shuffle.c
+++ b/blosc/shuffle.c
@@ -25,17 +25,17 @@
 /*  Include hardware-accelerated shuffle/unshuffle routines based on
     the target architecture. Note that a target architecture may support
     more than one type of acceleration!*/
-#if defined(SHUFFLE_AVX2_ENABLED)
+#if defined(SHUFFLE_USE_AVX2)
   #include "shuffle-avx2.h"
   #include "bitshuffle-avx2.h"
-#endif  /* defined(SHUFFLE_AVX2_ENABLED) */
+#endif  /* defined(SHUFFLE_USE_AVX2) */
 
-#if defined(SHUFFLE_SSE2_ENABLED)
+#if defined(SHUFFLE_USE_SSE2)
   #include "shuffle-sse2.h"
   #include "bitshuffle-sse2.h"
-#endif  /* defined(SHUFFLE_SSE2_ENABLED) */
+#endif  /* defined(SHUFFLE_USE_SSE2) */
 
-#if defined(SHUFFLE_NEON_ENABLED)
+#if defined(SHUFFLE_USE_NEON)
   #if defined(__linux__)
     #include <sys/auxv.h>
     #ifdef ARM_ASM_HWCAP
@@ -44,12 +44,12 @@
   #endif
   #include "shuffle-neon.h"
   #include "bitshuffle-neon.h"
-#endif  /* defined(SHUFFLE_NEON_ENABLED) */
+#endif  /* defined(SHUFFLE_USE_NEON) */
 
-#if defined(SHUFFLE_ALTIVEC_ENABLED)
+#if defined(SHUFFLE_USE_ALTIVEC)
   #include "shuffle-altivec.h"
   #include "bitshuffle-altivec.h"
-#endif  /* defined(SHUFFLE_ALTIVEC_ENABLED) */
+#endif  /* defined(SHUFFLE_USE_ALTIVEC) */
 
 
 /*  Define function pointer types for shuffle/unshuffle routines. */
@@ -84,7 +84,7 @@ typedef enum {
 
 /* Detect hardware and set function pointers to the best shuffle/unshuffle
    implementations supported by the host processor. */
-#if defined(SHUFFLE_AVX2_ENABLED) || defined(SHUFFLE_SSE2_ENABLED)    /* Intel/i686 */
+#if defined(SHUFFLE_USE_AVX2) || defined(SHUFFLE_USE_SSE2)    /* Intel/i686 */
 
 /*  Disabled the __builtin_cpu_supports() call, as it has issues with
     new versions of gcc (like 5.3.1 in forthcoming ubuntu/xenial:
@@ -250,7 +250,7 @@ static blosc_cpu_features blosc_get_cpu_features(void) {
 }
 #endif /* HAVE_CPU_FEAT_INTRIN */
 
-#elif defined(SHUFFLE_NEON_ENABLED) /* ARM-NEON */
+#elif defined(SHUFFLE_USE_NEON) /* ARM-NEON */
 static blosc_cpu_features blosc_get_cpu_features(void) {
   blosc_cpu_features cpu_features = BLOSC_HAVE_NOTHING;
 #if defined(__aarch64__)
@@ -263,7 +263,7 @@ static blosc_cpu_features blosc_get_cpu_features(void) {
 #endif
   return cpu_features;
 }
-#elif defined(SHUFFLE_ALTIVEC_ENABLED) /* POWER9-ALTIVEC preliminary test*/
+#elif defined(SHUFFLE_USE_ALTIVEC) /* POWER9-ALTIVEC preliminary test*/
 static blosc_cpu_features blosc_get_cpu_features(void) {
   blosc_cpu_features cpu_features = BLOSC_HAVE_NOTHING;
   cpu_features |= BLOSC_HAVE_ALTIVEC;
@@ -280,11 +280,11 @@ static blosc_cpu_features blosc_get_cpu_features(void) {
 return BLOSC_HAVE_NOTHING;
 }
 
-#endif /* defined(SHUFFLE_AVX2_ENABLED) || defined(SHUFFLE_SSE2_ENABLED) */
+#endif /* defined(SHUFFLE_USE_AVX2) || defined(SHUFFLE_USE_SSE2) */
 
 static shuffle_implementation_t get_shuffle_implementation(void) {
   blosc_cpu_features cpu_features = blosc_get_cpu_features();
-#if defined(SHUFFLE_AVX2_ENABLED)
+#if defined(SHUFFLE_USE_AVX2)
   if (cpu_features & BLOSC_HAVE_AVX2) {
     shuffle_implementation_t impl_avx2;
     impl_avx2.name = "avx2";
@@ -294,9 +294,9 @@ static shuffle_implementation_t get_shuffle_implementation(void) {
     impl_avx2.bitunshuffle = (bitunshuffle_func)bshuf_untrans_bit_elem_avx2;
     return impl_avx2;
   }
-#endif  /* defined(SHUFFLE_AVX2_ENABLED) */
+#endif  /* defined(SHUFFLE_USE_AVX2) */
 
-#if defined(SHUFFLE_SSE2_ENABLED)
+#if defined(SHUFFLE_USE_SSE2)
   if (cpu_features & BLOSC_HAVE_SSE2) {
     shuffle_implementation_t impl_sse2;
     impl_sse2.name = "sse2";
@@ -306,9 +306,9 @@ static shuffle_implementation_t get_shuffle_implementation(void) {
     impl_sse2.bitunshuffle = (bitunshuffle_func)bshuf_untrans_bit_elem_sse2;
     return impl_sse2;
   }
-#endif  /* defined(SHUFFLE_SSE2_ENABLED) */
+#endif  /* defined(SHUFFLE_USE_SSE2) */
 
-#if defined(SHUFFLE_NEON_ENABLED)
+#if defined(SHUFFLE_USE_NEON)
   if (cpu_features & BLOSC_HAVE_NEON) {
     shuffle_implementation_t impl_neon;
     impl_neon.name = "neon";
@@ -326,9 +326,9 @@ static shuffle_implementation_t get_shuffle_implementation(void) {
     impl_neon.bitunshuffle = (bitunshuffle_func)bshuf_untrans_bit_elem_scal;
     return impl_neon;
   }
-#endif  /* defined(SHUFFLE_NEON_ENABLED) */
+#endif  /* defined(SHUFFLE_USE_NEON) */
 
-#if defined(SHUFFLE_ALTIVEC_ENABLED)
+#if defined(SHUFFLE_USE_ALTIVEC)
   if (cpu_features & BLOSC_HAVE_ALTIVEC) {
     shuffle_implementation_t impl_altivec;
     impl_altivec.name = "altivec";
@@ -338,7 +338,7 @@ static shuffle_implementation_t get_shuffle_implementation(void) {
     impl_altivec.bitunshuffle = (bitunshuffle_func)bshuf_untrans_bit_elem_altivec;
     return impl_altivec;
   }
-#endif  /* defined(SHUFFLE_ALTIVEC_ENABLED) */
+#endif  /* defined(SHUFFLE_USE_ALTIVEC) */
 
   /* Processor doesn't support any of the hardware-accelerated implementations,
      so use the generic implementation. */

--- a/blosc/shuffle.h
+++ b/blosc/shuffle.h
@@ -22,6 +22,25 @@
 
 #include "blosc2/blosc2-common.h"
 
+/* Toggle hardware-accelerated routines based on SHUFFLE_*_ENABLED macros
+   and availability on the target architecture.
+*/
+#if defined(SHUFFLE_AVX2_ENABLED) && defined(__AVX2__)
+#define SHUFFLE_USE_AVX2
+#endif
+
+#if defined(SHUFFLE_SSE2_ENABLED) && defined(__SSE2__)
+#define SHUFFLE_USE_SSE2
+#endif
+
+#if defined(SHUFFLE_ALTIVEC_ENABLED) && defined(__ALTIVEC__)
+#define SHUFFLE_USE_ALTIVEC
+#endif
+
+#if defined(SHUFFLE_NEON_ENABLED) && defined(__ARM_NEON)
+#define SHUFFLE_USE_NEON
+#endif
+
 #ifdef __cplusplus
 extern "C" {
 #endif

--- a/tests/test_shuffle_roundtrip_altivec.c
+++ b/tests/test_shuffle_roundtrip_altivec.c
@@ -17,7 +17,7 @@
 
 /* Include ALTIVEC-accelerated shuffle implementation if supported by this compiler.
    TODO: Need to also do run-time CPU feature support here. */
-#if defined(SHUFFLE_ALTIVEC_ENABLED)
+#if defined(SHUFFLE_USE_ALTIVEC)
   #include "../blosc/shuffle-altivec.h"
 #else
   #if defined(_MSC_VER)
@@ -25,13 +25,13 @@
   #else
     #warning ALTIVEC shuffle tests not enabled.
   #endif
-#endif  /* defined(SHUFFLE_ALTIVEC_ENABLED) */
+#endif  /* defined(SHUFFLE_USE_ALTIVEC) */
 
 
 /** Roundtrip tests for the ALTIVEC-accelerated shuffle/unshuffle. */
 static int test_shuffle_roundtrip_altivec(int32_t type_size, int32_t num_elements,
                                           size_t buffer_alignment, int test_type) {
-#if defined(SHUFFLE_ALTIVEC_ENABLED)
+#if defined(SHUFFLE_USE_ALTIVEC)
   int32_t buffer_size = type_size * num_elements;
 
   /* Allocate memory for the test. */
@@ -77,7 +77,7 @@ static int test_shuffle_roundtrip_altivec(int32_t type_size, int32_t num_element
   return exit_code;
 #else
   return EXIT_SUCCESS;
-#endif /* defined(SHUFFLE_ALTIVEC_ENABLED) */
+#endif /* defined(SHUFFLE_USE_ALTIVEC) */
 }
 
 

--- a/tests/test_shuffle_roundtrip_avx2.c
+++ b/tests/test_shuffle_roundtrip_avx2.c
@@ -17,7 +17,7 @@
 /* Include accelerated shuffles if supported by this compiler.
    TODO: Need to also do run-time CPU feature support here. */
 
-#if defined(SHUFFLE_AVX2_ENABLED)
+#if defined(SHUFFLE_USE_AVX2)
   #include "../blosc/shuffle-avx2.h"
 #else
   #if defined(_MSC_VER)
@@ -25,13 +25,13 @@
   #else
     #warning AVX2 shuffle tests not enabled.
   #endif
-#endif  /* defined(SHUFFLE_AVX2_ENABLED) */
+#endif  /* defined(SHUFFLE_USE_AVX2) */
 
 
 /** Roundtrip tests for the AVX2-accelerated shuffle/unshuffle. */
 static int test_shuffle_roundtrip_avx2(int32_t type_size, int32_t num_elements,
                                        size_t buffer_alignment, int test_type) {
-#if defined(SHUFFLE_AVX2_ENABLED)
+#if defined(SHUFFLE_USE_AVX2)
   int32_t buffer_size = type_size * num_elements;
 
   /* Allocate memory for the test. */
@@ -78,7 +78,7 @@ static int test_shuffle_roundtrip_avx2(int32_t type_size, int32_t num_elements,
   return exit_code;
 #else
   return EXIT_SUCCESS;
-#endif /* defined(SHUFFLE_AVX2_ENABLED) */
+#endif /* defined(SHUFFLE_USE_AVX2) */
 }
 
 

--- a/tests/test_shuffle_roundtrip_neon.c
+++ b/tests/test_shuffle_roundtrip_neon.c
@@ -17,7 +17,7 @@
 
 /* Include NEON-accelerated shuffle implementation if supported by this compiler.
    TODO: Need to also do run-time CPU feature support here. */
-#if defined(SHUFFLE_NEON_ENABLED)
+#if defined(SHUFFLE_USE_NEON)
   #include "../blosc/shuffle-neon.h"
 #else
   #if defined(_MSC_VER)
@@ -25,13 +25,13 @@
   #else
     #warning NEON shuffle tests not enabled.
   #endif
-#endif  /* defined(SHUFFLE_NEON_ENABLED) */
+#endif  /* defined(SHUFFLE_USE_NEON) */
 
 
 /** Roundtrip tests for the NEON-accelerated shuffle/unshuffle. */
 static int test_shuffle_roundtrip_neon(size_t type_size, size_t num_elements,
                                        size_t buffer_alignment, int test_type) {
-#if defined(SHUFFLE_NEON_ENABLED)
+#if defined(SHUFFLE_USE_NEON)
   size_t buffer_size = type_size * num_elements;
 
   /* Allocate memory for the test. */
@@ -77,7 +77,7 @@ static int test_shuffle_roundtrip_neon(size_t type_size, size_t num_elements,
   return exit_code;
 #else
   return EXIT_SUCCESS;
-#endif /* defined(SHUFFLE_NEON_ENABLED) */
+#endif /* defined(SHUFFLE_USE_NEON) */
 }
 
 

--- a/tests/test_shuffle_roundtrip_sse2.c
+++ b/tests/test_shuffle_roundtrip_sse2.c
@@ -17,7 +17,7 @@
 
 /* Include SSE2-accelerated shuffle implementation if supported by this compiler.
    TODO: Need to also do run-time CPU feature support here. */
-#if defined(SHUFFLE_SSE2_ENABLED)
+#if defined(SHUFFLE_USE_SSE2)
   #include "../blosc/shuffle-sse2.h"
 #else
   #if defined(_MSC_VER)
@@ -25,13 +25,13 @@
   #else
     #warning SSE2 shuffle tests not enabled.
   #endif
-#endif  /* defined(SHUFFLE_SSE2_ENABLED) */
+#endif  /* defined(SHUFFLE_USE_SSE2) */
 
 
 /** Roundtrip tests for the SSE2-accelerated shuffle/unshuffle. */
 static int test_shuffle_roundtrip_sse2(int32_t type_size, int32_t num_elements,
                                        size_t buffer_alignment, int test_type) {
-#if defined(SHUFFLE_SSE2_ENABLED)
+#if defined(SHUFFLE_USE_SSE2)
   int32_t buffer_size = type_size * num_elements;
 
   /* Allocate memory for the test. */
@@ -77,7 +77,7 @@ static int test_shuffle_roundtrip_sse2(int32_t type_size, int32_t num_elements,
   return exit_code;
 #else
   return EXIT_SUCCESS;
-#endif /* defined(SHUFFLE_SSE2_ENABLED) */
+#endif /* defined(SHUFFLE_USE_SSE2) */
 }
 
 

